File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/keys/SeedValidationException.java
package com.github.dedis.popstellar.utility.error.keys;

import com.github.dedis.popstellar.R;

public class SeedValidationException extends KeyException {

  public SeedValidationException(Exception cause) {
    super("Unable to validate given seed", cause);
  }

  @Override
  public int getUserMessage() {
    return R.string.seed_validation_exception;
  }

  @Override
  public Object[] getUserMessageArguments() {
    return new Object[0];
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/keys/SeedValidationException.kt
package com.github.dedis.popstellar.utility.error.keys

import com.github.dedis.popstellar.R

class SeedValidationException(cause: Exception?) :
    KeyException("Unable to validate given seed", cause) {
    override val userMessage: Int
        get() = R.string.seed_validation_exception
    override val userMessageArguments: Array<Any?>
        get() = arrayOfNulls(0)
}